web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -45,14 +49,25 @@
 
     @Autowired
     private SampleService sampleService;
+    
+    @Autowired
+    private StudyService studyService;
 
     @Autowired
     private UniqueKeyExtractor uniqueKeyExtractor;
 
-    @PreAuthorize("hasPermission(#studyId, 'CancerStudyId', 'read')")
+    @Value("${authenticate:false}")
+    private String authenticate;
+    
+    private boolean usingAuth() {
```

Comment:
this is the fourth implementation of logic determining whether the portal is running in an "open to the public" or an authenticated / authorized access mode. We should soon create a shared utility function which can be called here and other places (if possible even during the wiring of applicationContext-security). (see core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/util/CacheMapUtil.java, security/security-spring/src/main/resources/applicationContext-security.xml)

Reply:
Leaving this off the PR, but yes, agree.